simphony/io/tests/test_file_mesh.py

```diff
@@ -375,6 +375,93 @@ def test_get_all_cells_iterator(self):
 
         self.assertItemsEqual(icells_id, cuids)
 
+    def test_get_all_edges_iterator_with_data(self):
+        """ Checks the edge iterator
+
+        Checks that an interator over all
+        the edges of the mesh is returned
+        when the function iter_edges is called
+        without arguments and its data is correctly retrieved
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:3]]
+
+        edges = [
+            Edge(puids[0:2], data=DataContainer({CUBA.VELOCITY: [0, 0, 0]}))
```

Comment:
I would suggest to use immutable objects like tuples for the `DataContainer` values. Especially tuples are faster to create and take much less memory resources than lists.